### PR TITLE
Fix asserts: name not in search results (#7278)

### PR DESCRIPTION
**Pull request by swadeley**
Hello

I changed all asserts after a delete as follows:

             session.syncplan.delete(plan_name)
    -        assert not session.syncplan.search(plan_name)
    +        assert 'Your search returned zero Sync Plans.' in str(session.syncplan.search(plan_name))

For scoped search, which does not delte the plan, it just should be there:

    -        assert not session.syncplan.search('enabled = false')
    +        assert 'Your search returned zero Sync Plans.' in str(session.syncplan.search('enabled = false'))

### CODE REVIEW

File: tests/foreman/ui/test_syncplan.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def test_positive_end_to_end(session):
         assert syncplan_values['details']['description'] == new_description
         # Delete sync plan
         session.syncplan.delete(plan_name)
-        assert not session.syncplan.search(plan_name)
+        assert 'Your search returned zero Sync Plans.' in str(session.syncplan.search(plan_name))
```
Comment (omkarkhatavkar) [inline tests/foreman/ui/test_syncplan.py]: `empty_search message = 'Your search returned zero Sync Plans.'` at module level and use across other tests if it is common. Just suggestion.

File: tests/foreman/ui/test_syncplan.py [hunk 2]
```diff
@@ -171,7 +171,7 @@ def test_positive_end_to_end_custom_cron(session):
         assert not syncplan_values['details']['cron_expression']
         # Delete sync plan
         session.syncplan.delete(plan_name)
-        assert not session.syncplan.search(plan_name)
+        assert 'Your search returned zero Sync Plans.' in str(session.syncplan.search(plan_name))
```
Comment (omkarkhatavkar) [inline tests/foreman/ui/test_syncplan.py]: Why every time you are converting to str object?

File: tests/foreman/ui/test_syncplan.py (hunk 1 above)
Comment (swadeley) [reply tests/foreman/ui/test_syncplan.py]: Do you mean something like:

    @fixture(scope='module')
    def module_org():
    global empty_search_message
    empty_search_message = 'Your search returned zero Sync Plans.'
    return entities.Organization().create()

File: tests/foreman/ui/test_syncplan.py (hunk 2 above)
Comment (swadeley) [reply tests/foreman/ui/test_syncplan.py]: The is one assert at the end of every test. Maybe I misunderstand?

Comment (omkarkhatavkar) [reply tests/foreman/ui/test_syncplan.py]: I was talking about `str(session.syncplan.search(plan_name))`. why there explicitly covert object into a string.   

if this not works 
`assert 'Your search returned zero Sync Plans.' in session.syncplan.search(plan_name)`

try below ones ?

`assert 'Your search returned zero Sync Plans.' in session.syncplan.search(plan_name)[0]['Name']`
or 
`session.syncplan.search(plan_name)[0]['Name'] != plan_name`

Comment (swadeley) [reply tests/foreman/ui/test_syncplan.py]: OK, I did try some of those before, trying again with 1:

    >           assert empty_search_message in session.syncplan.search(plan_name)
    E           AssertionError: assert 'Your search returned zero Sync Plans.' in [{'Description': 'Your search returned zero Sync Plans.', 'Interval': 'Your search returned zero Sync Plans.','Name': 'Your search returned zero Sync Plans.', 'Next Sync': 'Your search returned zero Sync Plans.', ...}]

2nd one works, and 3rd one works:
    assert session.syncplan.search(plan_name)[0]['Name'] != plan_name

But there is this to consider:
https://github.com/SatelliteQE/robottelo/issues/7272

Comment (swadeley) [reply tests/foreman/ui/test_syncplan.py]: This method works, and complies with https://github.com/SatelliteQE/robottelo/issues/7272


    @@ -173,7 +173,7 @@ def test_positive_end_to_end_custom_cron(session):
         assert not syncplan_values['details']['cron_expression']
         # Delete sync plan
         session.syncplan.delete(plan_name)
    -        assert empty_search_message in str(session.syncplan.search(plan_name))
    +        assert plan_name not in session.syncplan.search(plan_name)
 
 
     @tier2
    @@ -209,8 +209,7 @@ def test_positive_search_scoped(session):
         ]:
             assert session.syncplan.search(
                 '{} = {}'.format(query_type, query_value))[0]['Name'] == name
    -        assert empty_search_message in str(
    -            session.syncplan.search('enabled = false'))
    +        assert name not in session.syncplan.search('enabled = false')

Comment (swadeley) [reply tests/foreman/ui/test_syncplan.py]: Changes made, will change pull request title to match latest change.

Comment (swadeley) [reply tests/foreman/ui/test_syncplan.py]: I could run a test with "not" removed from those asserts just to make sure it fails.

**Comment by swadeley** [comment]
robottelo]$ pytest --show-capture=no tests/foreman/ui/test_syncplan.py
2019-08-15 09:07:36 - conftest - DEBUG - Registering custom pytest_configure

===================== test session starts ===================================
platform linux -- Python 3.6.6, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /sat6/robottelo
plugins: services-1.3.1, forked-1.0.2, xdist-1.29.0, cov-2.7.1, mock-1.10.4
collecting ... 2019-08-15 09:07:37 - conftest - DEBUG - BZ deselect is disabled in settings

collected 5 items                                                                                                                                            

tests/foreman/ui/test_syncplan.py .....                                                                           [100%]

================= 5 passed, 9 warnings in 864.12 seconds ======================

File: tests/foreman/ui/test_syncplan.py (hunk 2 above)
Comment (swadeley) [reply tests/foreman/ui/test_syncplan.py]: I did that, removing the "not" makes the assert fail.

**Comment by swadeley** [comment]
After running the test I had to add one empty line, not squashed yet in interests of transparency. Merger please squash on merge.

Thank you

**Comment by swadeley** [comment]
Thank you all

**Pull request merged**